logbook-servlet/src/main/java/org/zalando/logbook/servlet/LocalResponse.java

```diff
@@ -106,42 +114,54 @@ public PrintWriter getWriter() throws IOException {
         return body;
     }
 
-    private final class TeeServletOutputStream extends ServletOutputStream {
+    private static final class TeeServletOutputStream extends ServletOutputStream {
 
-        private final OutputStream original;
+        private final DataOutput output;
+        private final LazyStream original;
 
-        private TeeServletOutputStream() throws IOException {
-            this.original = LocalResponse.super.getOutputStream();
+        private TeeServletOutputStream(final DataOutput output,
+                final LazyStream original) {
```

Comment:
The whole point of this supplier is to access the stream lazily. That's the cause of this bug actually, since we access it too early we currently fail.